spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -152,84 +153,141 @@ public Subscriber subscribe(String subscription,
 	public PubsubMessage pullNext(String subscription) {
 		List<PubsubMessage> receivedMessageList = pullAndAck(subscription, 1, true);
 
-		return receivedMessageList.size() > 0 ?	receivedMessageList.get(0) : null;
+		return receivedMessageList.size() > 0 ? receivedMessageList.get(0) : null;
 	}
 
 	public SubscriberFactory getSubscriberFactory() {
 		return this.subscriberFactory;
 	}
 
 	@Override
-	public void ack(Collection<AcknowledgeablePubsubMessage> acknowledgeablePubsubMessages) {
-		Assert.notEmpty(acknowledgeablePubsubMessages, "The acknowledgeablePubsubMessages can't be null.");
+	public ListenableFuture<Void> ack(
+			Collection<AcknowledgeablePubsubMessage> acknowledgeablePubsubMessages) {
+		Assert.notEmpty(acknowledgeablePubsubMessages, "The acknowledgeablePubsubMessages cannot be null.");
```

Comment:
It's not about `null`. It's really `notEmpty` check.

Reply:
Agreed - will fix.